src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/ad/VideoFanTakeover.java

```diff
@@ -44,15 +47,20 @@ public void play() {
     waitForVideoStart();
   }
 
+  public void playOnFandom() {
+    runInAdFrame(() -> wait.forElementClickable(playTriggerButtonSelector).click());
+    waitForVideoStartOnFandom();
+  }
+
   public void pause() {
-    driver.findElement(By.className("overVideoLayer")).click();
+    driver.findElement(By.cssSelector(String.format(UI_ELEMENT_SELECTOR_FORMAT, slotName))).click();
   }
 
-  private interface Lambda {
-    void run();
+  public void pauseVideoOnFandom() {
+    driver.findElement(By.cssSelector(String.format(UI_ELEMENT_SELECTOR_ON_FANDOM_FORMAT, slotName))).click();
   }
 
-  private void runInAdFrame(Lambda f) {
+  private void runInAdFrame(Runnable f) {
     driver.switchTo().frame(iframe);
     f.run();
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Call the method Thread.start() to execute the content of the run() method in a dedicated thread. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS1217)